src/overlay/Peer.cpp

```diff
@@ -239,6 +240,13 @@ Peer::recurrentTimerExpired(asio::error_code const& error)
             drop("idle timeout", Peer::DropDirection::WE_DROPPED_REMOTE,
                  Peer::DropMode::IGNORE_WRITE_QUEUE);
         }
+        else if (flowControlEnabled() == Peer::FlowControlState::ENABLED &&
+                 (now - mLastSendMoreReceived) >= timeout)
```

Comment:
Is this correct? If the rate of messages is low and the capacity is high, we may not receive "send more" requests.
Maybe what we should be tracking is how long a connection is stuck "at 0 capacity" instead?

Reply:
yeah I think you're right: while it seems unlikely, it's possible, especially on testnet. I updated the change to start counting when the outbound capacity is 0 on the sender side. This way we don't depend on message rate, and also handle potential bugs in capacity accounting on the receiver side (for example, if it releases _some_ capacity, but not enough to trigger sending "send more").